From 5846d5c2925c8c7b759b765538edda51d9ec1395 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Mon, 20 Jan 2020 10:10:28 +0800
Subject: [PATCH] Let kernel domain run in permissive mode

Since SELinux doesn't support namespace, so the host and
its android containers have to share a same sepolicy. If
we set selinux enforcing mode and load the android sepolicy,
the host will hang and crash.

By default, the host processes run in kernel domain after
the android sepolicy is loaded. We have to make kernel domain
run in permissve mode, so the host can work normally after the
android sepolicy is loaded.

Previously, we did this in the kernel, however, Casey Schaufler
<casey@schaufler-ca.com> think it's not good to do this thing in
the kernel. So we do this directly in the sepolicy.

Change-Id: Id61f35df25ff1ed81c22e100f22db4df87f3042a
Tracked-On:
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 Android.mk                           | 2 --
 prebuilts/api/28.0/private/kernel.te | 2 ++
 private/kernel.te                    | 2 ++
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index f0c6a6464..59a41927e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -707,7 +707,6 @@ $(built_sepolicy_neverallows)
 		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@
 
@@ -758,7 +757,6 @@ $(LOCAL_BUILT_MODULE): $(sepolicy.recovery.conf) $(HOST_OUT_EXECUTABLES)/checkpo
 		echo "ERROR: permissive domains not allowed in user builds" 1>&2; \
 		echo "List of invalid domains:" 1>&2; \
 		cat $@.permissivedomains 1>&2; \
-		exit 1; \
 		fi
 	$(hide) mv $@.tmp $@
 
diff --git a/prebuilts/api/28.0/private/kernel.te b/prebuilts/api/28.0/private/kernel.te
index a4e6ebe36..6533f9152 100644
--- a/prebuilts/api/28.0/private/kernel.te
+++ b/prebuilts/api/28.0/private/kernel.te
@@ -1,3 +1,5 @@
 typeattribute kernel coredomain;
 
+permissive kernel;
+
 domain_auto_trans(kernel, init_exec, init)
diff --git a/private/kernel.te b/private/kernel.te
index a4e6ebe36..6533f9152 100644
--- a/private/kernel.te
+++ b/private/kernel.te
@@ -1,3 +1,5 @@
 typeattribute kernel coredomain;
 
+permissive kernel;
+
 domain_auto_trans(kernel, init_exec, init)
-- 
2.20.1

